tests/Unit/Helpers/ParallelAlgorithms/Interpolation/InterpolateOnElementTestHelpers.hpp

```diff
@@ -149,114 +165,174 @@ struct mock_interpolation_target {
-        interp_point_info_l) = block_logical_coordinates(domain, target_points);
-    return interp_point_info_l;
-  }();
+    auto& runner = *runner_ptr;
 
-  // Emplace target component.
-  ActionTesting::MockRuntimeSystem<metavars> runner{
-      tuples::tagged_tuple_from_typelist<
-          Parallel::get_const_global_cache_tags<metavars>>{
-          std::move(global_cache_items)...}};
-  ActionTesting::set_phase(make_not_null(&runner),
-                           metavars::Phase::Initialization);
-  ActionTesting::emplace_component_and_initialize<target_component>(
-      &runner, 0, {target_points});
+    ActionTesting::set_phase(make_not_null(&runner),
+                             metavars::Phase::Initialization);
+    ActionTesting::emplace_component_and_initialize<target_component>(
+        &runner, 0, {target_points});
 
-  static_assert(
-      std::is_same_v<typename metavars::InterpolationTargetA::temporal_id::type,
-                     double> or
-          std::is_same_v<
-              typename metavars::InterpolationTargetA::temporal_id::type,
-              TimeStepId>,
-      "Unsupported temporal_id type");
-  if constexpr (std::is_same_v<
-                    typename metavars::InterpolationTargetA::temporal_id::type,
-                    double>) {
-    initialize_elements_and_queue_simple_actions(
-        domain_creator, domain, element_ids, interp_point_info, runner,
-        temporal_id.substep_time().value());
-  } else if constexpr (std::is_same_v<typename metavars::InterpolationTargetA::
-                                          temporal_id::type,
-                                      TimeStepId>) {
-    initialize_elements_and_queue_simple_actions(domain_creator, domain,
-                                                 element_ids, interp_point_info,
-                                                 runner, temporal_id);
-  }
+    static_assert(
+        std::is_same_v<
+            typename metavars::InterpolationTargetA::temporal_id::type,
+            double> or
+            std::is_same_v<
+                typename metavars::InterpolationTargetA::temporal_id::type,
+                TimeStepId>,
+        "Unsupported temporal_id type");
+    if constexpr (std::is_same_v<typename metavars::InterpolationTargetA::
+                                     temporal_id::type,
+                                 double>) {
+      initialize_elements_and_queue_simple_actions(
+          *domain_creator, domain, element_ids, interp_point_info, runner,
+          temporal_id.substep_time().value());
+    } else if constexpr (std::is_same_v<
+                             typename metavars::InterpolationTargetA::
+                                 temporal_id::type,
+                             TimeStepId>) {
+      initialize_elements_and_queue_simple_actions(
+          *domain_creator, domain, element_ids, interp_point_info, runner,
+          temporal_id);
+    }
```

Comment:
Can you replace all this with `intrp::InterpolationTarget_detail::get_temporal_id_value`?

Reply:
No, that doesn't work (unless there are changes elsewhere).  Here `temporal_id` is always a `TimeStepId`, but the last argument of `initialize_elements_and_queue_simple_actions` needs to be the same type as `metavars::InterpolationTargetA::temporal_id::type` which might not be a `TimeStepId`.